### PR TITLE
Update changelog for v0.15.0, bump cargo.toml

### PR DESCRIPTION
**Pull request by clarfonthey**
Rather than just asking @Amanieu to do all this, I decided to help out a bit. Put some question marks in the date field that will need to be replaced once this is actually merged.

Will still need someone to do a proper release for this, but hopefully the changelog is good at least.

### CODE REVIEW

**Comment by clarfonthey** [comment]
cc @Amanieu @cuviper I updated this to include everything that's been merged plus #556 since I figured you wanted to get that in before 0.15 too. Anything else that needs fixing before release?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,9 +7,42 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
...
+- `ahash` feature has been renamed to `default-hasher`. (#533)
+- Entry API has been reworked and several methods have been renamed. (#535)
+- `Hash{Map,Set}::insert_unique_unchecked` is now unsafe. (#556)
+
+### Fixed
+
+* Fixed typos, stray backticks in docs. (#558, #560)
+
+### Removed
+
```
Comment (cuviper) [inline CHANGELOG.md:41]:
```suggestion

- `HashSet::get_or_insert_owned` has been removed in favor of `get_or_insert_with`. (#555)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,9 +7,42 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
...
+centralising on the `HashTable` API in the future. You can follow the discussion
+and progress in #545 to discuss features you think should be added to this API
+that were previously only possible on the `raw` API.
+
+### Added
+
+- Added `borsh` feature with `BorshSerialize` and `BorshDeserialize` impls. (#525)
+- Added `Assign` impls for `HashSet` operators. (#529)
+- Added `Default` impls for iterator types. (#542)
+- Added `RawTable::iter_hash` and `HashTable::iter_hash{,_mut}` methods. (#549)
```
[the remark this answers is not in the document]
Comment (clarfonthey) [reply CHANGELOG.md:22]: Right. 🤦🏻

**Comment by Amanieu** [comment]
With #561 and #556 merged we should be good to release.

**Comment by clarfonthey** [comment]
Everything should hopefully be up to date now.

**Comment by Amanieu** [comment]
Finally, we're ready for release!

@bors r+

**Pull request merged**